pspipe_utils/best_fits.py

```diff
@@ -175,33 +189,37 @@ def get_all_best_fit(spec_name_list, l_th, cmb_dict, fg_dict, spectra, nl_dict=N
     ps_all_th, nl_all_th = {}, {}
 
     for spec_name in spec_name_list:
-        na, nb = spec_name.split("x")
-        if len(na.split("&")) == 2:
-            sv_a, ar_a = na.split("&")
-            sv_b, ar_b = nb.split("&")
+        ms_a, ms_b = spec_name.split("x")
+        if len(ms_a.split(char)) == 2:
+            sv_a, ar_a = ms_a.split(char)
+            sv_b, ar_b = ms_b.split(char)
             noise_key_a = ar_a
             noise_key_b = ar_b
-        elif len(na.split("&")) == 3:
-            sv_a, ar_a, split_a = na.split("&")
-            sv_b, ar_b, split_b = nb.split("&")
+        elif len(ms_a.split("&")) == 3:
```

Comment:
here you should also used the parameter `char` when splitting. You can also write this part in a simpler way
```python
tuple_a = ms_a.split(char)
tuple_b = ms_b.split(char)
if len(tuple_a) == 2:
  sv_a, ar_a = tuple_a
  sv_b, ar_b = tuple_b
  ...
elif len(tuple_a) == 3:
  sv_a, ar_a, split_a = tuple_a
else:
  raise ValueError()
```
I should also add an exception in case the split does not work as expected, to understand what's going on